### PR TITLE
Product and Category models, services, repositories and controllers i…

### PR DESCRIPTION
**Pull request by hpavlo**
…mplemented

### CODE REVIEW

**Review by YuriiHryniv — APPROVED** [review]
Looks good to me, but I would have added Lombok annotations for boilerplate code in models and dtos

File: src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("categories")
```
Comment (Noct2000) [inline src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java:19]:
```suggestion
@RequestMapping("/categories")
```

File: src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productRequestDtoMapper.mapToModel(productRequestDto);
+        product.setId(id);
+        return productResponseDtoMapper.mapToDto(productService.update(product));
+    }
+
+    @GetMapping("byPrice")
```
Comment (Noct2000) [inline src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java:63]:
```suggestion
    @GetMapping("/byPrice")
```

File: src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    public List<ProductResponseDto> getAllByPrice(@RequestParam BigDecimal from,
+                                                  @RequestParam BigDecimal to) {
+        return productService.getAllByPrice(from, to)
+                .stream()
+                .map(productResponseDtoMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping()
```
Comment (Noct2000) [inline src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java:72]:
```suggestion
    @GetMapping
```

File: src/main/java/mate/academy/springboot/datajpa/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package mate.academy.springboot.datajpa.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "category")
```
Comment (Noct2000) [inline src/main/java/mate/academy/springboot/datajpa/model/Category.java:10]:
```suggestion
@Table(name = "categories")
```

File: src/main/java/mate/academy/springboot/datajpa/model/Product.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+
+@Entity
+@Table(name = "product")
```
Comment (Noct2000) [inline src/main/java/mate/academy/springboot/datajpa/model/Product.java:12]:
```suggestion
@Table(name = "products")
```

**Review by Noct2000 — CHANGES_REQUESTED** [review]

**Review by Noct2000 — APPROVED** [review]

**Pull request closed without merging**